core/trino-main/src/main/java/io/trino/operator/HashAggregationOperator.java

```diff
@@ -368,8 +385,14 @@ public void addInput(Page page)
         inputProcessed = true;
 
         if (aggregationBuilder == null) {
-            // TODO: We ignore spillEnabled here if any aggregate has ORDER BY clause or DISTINCT because they are not yet implemented for spilling.
-            if (step.isOutputPartial() || !spillEnabled || !isSpillable()) {
+            boolean partialAggregationDisabled = partialAggregationController
+                    .map(PartialAggregationController::isPartialAggregationDisabled)
+                    .orElse(false);
+            if (step.isOutputPartial() && partialAggregationDisabled) {
+                aggregationBuilder = new SkipAggregationBuilder(groupByChannels, hashChannel, aggregatorFactories, memoryContext);
```

Comment:
future improvement. It would be great to actually collect metrics
* how many pages were processed via skip aggregation builder
* how many flushes there were for PA. 
* what average row count per flush
* etc..

This can be returned via `io.trino.operator.OperatorContext#setLatestMetrics`

@lukasz-stec Maybe create an issue for that?